modules/hal_nordic/Kconfig

```diff
@@ -204,7 +204,7 @@ config NRF_802154_SECURITY_KEY_STORAGE_SIZE
 
 config NRF_802154_CARRIER_FUNCTIONS
 	bool "nRF 802.15.4 carrier functions"
-	default y if OPENTHREAD_DIAG
+	default y if (OPENTHREAD_DIAG || IEEE802154_CARRIER_FUNCTIONS)
```

Comment:
as `OPENTHREAD_DIAG` now depends on `IEEE802154_CARRIER_FUNCTIONS` isn't this or statement redundant? We will never have a situation when diag is enabled but carrier functions is disabled. so maybe just `default y if IEEE802154_CARRIER_FUNCTIONS` would be enough?